### PR TITLE
adjust signer/authority requirements

### PR DESCRIPTION
**Pull request by suscd**
- creating an auth account is now permissionless (not requiring the target owner of the account to sign in order to create it)
- enforce authority addresses in builds by default, require explicit `testing` feature to opt out

- minor lint updates

### CODE REVIEW

File: programs/rewards/src/instructions/airdrop_close.rs [hunk 1]
```diff
@@ -17,6 +16,7 @@ pub struct AirdropClose<'info> {
     pub reward_vault: Account<'info, TokenAccount>,
 
     /// The authority to make changes to the airdrop, which must sign
+    #[cfg_attr(not(feature = "testing"), account(address = GOVERNOR_ID))]
```
Comment (dnut) [inline programs/rewards/src/instructions/airdrop_close.rs:19]: So, anyone is allowed to create an airdrop, with any authority, but only the governor can close it? This seems a little weird at first glance. But I think I see the purpose.

Anyone can easily construct an airdrop on the client side without the inconvenience of going through governance. There's no risk in letting anyone do that because governance has to approve an actual funding of that airdrop, and once the airdrop is finalized the creator relinquishes all power over it. Presumably, the funding proposal would only pass after the airdrop is finalized. Once funded, you don't want to need to trust the person who authored the proposal to avoid siphoning all the funds into their wallet. Only the governance that funded it should be allowed to do that. Does this sound right?

I think the distinction between the two roles is unclear because of ambiguous naming. You could rename the authority field in the account to "author" and rename the field here to "close_authority".

You could even include a "close_authority" field that is added to the airdrop account at the time of creation. This constraint could be moved there. But actually I think it would be better to always let the creator choose an arbitrary value for the close_authority. Then we have no need for conditional compilation or hardcoding any specific governances into the program. The governance can decide not to fund the account if they don't like how the close_authority is configured.

I also think it would be valuable for jetctl to inspect funding instructions for airdrops, so voters can easily verify that the airdrop is finalized and that the close_authority is set correctly.

You did not apply similar logic to awards and distributions. For awards, I guess that's because the funder and creator are the same account, so there cannot be a conflict of interest there. For distributions, I guess it's because you can only close it after it's fully paid out. Why not allow anyone to close it? or automatically close it once it is fully paid out?

File: programs/rewards/Cargo.toml [hunk 2]
```diff
@@ -14,11 +14,13 @@ no-entrypoint = []
 no-idl = []
 cpi = ["no-entrypoint"]
 default = []
+testing = []
 cli = ["no-entrypoint", "serde"]
 
 [dependencies]
 anchor-lang = { git = "https://github.com/jet-lab/anchor", branch = "master" }
 anchor-spl = { git = "https://github.com/jet-lab/anchor", branch = "master" }
+solana-program = "1.10"
```
Comment (dnut) [inline programs/rewards/Cargo.toml:23]: Why is this here?

Comment (suscd) [reply programs/rewards/Cargo.toml:23]: for `pubkey!` macro

File: programs/rewards/src/instructions/airdrop_close.rs (hunk 1 above)
Comment (dnut) [reply programs/rewards/src/instructions/airdrop_close.rs:19]: > I was originally implementing it this way, but the problem with this approach is that it's just not backwards compatible. There's an existing airdrop that also needs to be closed, so hardcoding it is a very simple fix.

The key that was used to create the airdrop is lost? It could have Pubkey::default() resolve to a default governance id, but that's kind of an awkward compromise.

> You don't even have to verify the authority as a reviewer if its hardcoded!

Well I think the verification is still there, you just need to verify that your governance works with the program instead of verifying that your governance works with the proposal. This can enable voters to verify it less often as long as they are aware of any program changes.

Comment (dnut) [reply programs/rewards/src/instructions/airdrop_close.rs:19]: > The key that was used to create the airdrop is lost?

If we have access to the key, my inclination would be to implement it as a field in the airdrop. If we do not have access to the key, I'm fine with the current approach.

Comment (suscd) [reply programs/rewards/src/instructions/airdrop_close.rs]: got the key to close the existing airdrop, but backing this change out from this PR, since it's not as urgent
